src/code_80005FD0.c

```diff
@@ -2751,21 +2751,21 @@ s16 func_8000D6D0(Vec3f arg0, s16 *waypointIndex, f32 arg2, f32 arg3, s16 pathIn
     zdiff = midZ - temp_f24;
     distance = sqrtf((xdiff * xdiff) + (ydiff * ydiff) + (zdiff * zdiff));
     if (distance > 0.01f) {
-        var_f2  = ((xdiff * arg2) / distance) + temp_f20;
-        var_f12 = ((ydiff * arg2) / distance) + temp_f22;
-        var_f14 = ((zdiff * arg2) / distance) + temp_f24;
+        var_f2  = ((xdiff * speed) / distance) + temp_f20;
+        var_f12 = ((ydiff * speed) / distance) + temp_f22;
+        var_f14 = ((zdiff * speed) / distance) + temp_f24;
     } else {
         var_f2  = temp_f20;
         var_f12 = temp_f22;
         var_f14 = temp_f24;
     }
-    arg0[0] = var_f2;
-    arg0[1] = var_f12;
-    arg0[2] = var_f14;
-    return func_80005FD0(sp50, arg0);
+    position[0] = var_f2;
+    position[1] = var_f12;
+    position[2] = var_f14;
+    return get_angle_path(sp50, position);
```

Comment:
sp50 good name might be `pos` or `oldPos`

get_angle_path if it returns the angle between sp50 and position. get_angle


get_angle_path is essentially `get_angle_between_two_vectors_dup`.

Perhaps `get_angle_between_two_vectors_vehicles`?

Reply:
yeah get_angle_path is a kind of duplicate of get_angle_between_two_vectors_dup